app/models/manageiq/providers/azure/cloud_manager/scanning/job.rb

```diff
@@ -20,6 +20,10 @@ def before_scan
     signal(:start_snapshot)
   end
 
+  def after_scan
+    signal(:snapshot_delete)
+  end
```

Comment:
This wasn't being overridden so the snapshot was never being deleted in the success case (which coincidentally never happened :laughing:)